### PR TITLE
feat(progress-indicator): adds hidden span to announce state of completion

### LINKED ISSUE #4246: AVT 3 - Progress Indicator has several screen reader issues
## Environment
macOS Mojave version 10.14.6
Chrome Version Version 77.0.3865.90
Carbon v10 - React

## Detailed Description

1. Go to the Progress Indicator Default Component
2. Start VoiceOver
3. Control-Option-Shift to enter the Web area. VoiceOver announces you are in a list with 5 items.
4. All items in the list are read as buttons, but only steps 2 & 3 are in the tab order.

Expected Result:
1. All buttons should be in the tab order and the user advised that its a link.
2. The state of the first button would indicate that it is complete.
3. The state of the second button would indicate it is the current step. 
4. The state of the third step as not started or incomplete.
5. The fifth button should let the user know that it is disabled when read.

### PR DESCRIPTION
**Pull request by abbeyhrt**
Closes #4246 

This PR adds a hidden span that will announce the state of the step, "Complete", "current", or "invalid" for screenreaders. 

#### Changelog

**New**

- `span` announcing step completion 

**Changed**

- `div` with `role="button"` to a `button` element 
- button reset and styling 
- update tests to not target `role="button"`

#### Testing / Reviewing

Using whatever screenreader you prefer to test the component and ensure that the completion state is announced.

### CODE REVIEW

File: packages/react/src/components/ProgressIndicator/ProgressIndicator.js [hunk 1]
```diff
@@ -70,16 +70,31 @@ export const ProgressStep = ({ ...props }) => {
     );
   };
 
+  let message = 'Incomplete';
```
Comment (asudoh) [inline packages/react/src/components/ProgressIndicator/ProgressIndicator.js]: g11n; Do we want to add a prop to let app specify the assistive text? (Though translate-ability for assistive text didn't use to be a hard-requirements, I have seen several issues asking for it)

**Review by asudoh — COMMENTED** [review]
Thank you for this fix @abbeyhrt!

**Review by joshblack — APPROVED** [review]
Seems great, also totally understand if you want to add support for translation per the feedback above 👍

**Review by tw15egan — APPROVED** [review]
LGTM! 👍 ✅

File: packages/react/src/components/ProgressIndicator/ProgressIndicator.js (hunk 1 above)
Comment (abbeyhrt) [reply packages/react/src/components/ProgressIndicator/ProgressIndicator.js]: Good idea! I added it, it's the first time I've done it so let me know if it all looks good!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @abbeyhrt!

**Pull request merged**